Issue #2894259 by WidgetsBurritos: Validate Sitemap URL on Submit

WidgetsBurritos opened the pull request:
Drupal Issue: https://www.drupal.org/node/2894259

This PR validates urls and sitemaps on submission.

WidgetsBurritos committed:
Issue #2894259 by WidgetsBurritos: Validate Sitemap URL on Submit

bighappyface commented:
@WidgetsBurritos please see the status check failure related to code style

bighappyface commented on src/Form/WebPageArchiveEditForm.php:
```diff
@@ -31,11 +32,13 @@ class WebPageArchiveEditForm extends WebPageArchiveFormBase {
    */
-  public function __construct(EntityStorageInterface $web_page_archive_storage, CaptureUtilityManager $capture_utility_manager) {
-    parent::__construct($web_page_archive_storage);
+  public function __construct(EntityStorageInterface $web_page_archive_storage, SitemapParser $sitemap_parser, CaptureUtilityManager $capture_utility_manager) {
```
How about making the `SitemapParser` the third parameter with a null default? Sitemap usage is optional and making it a required param within method parameters seems heavy. The same would apply to `WebPageArchiveFormBase` but the parameter is already at the end, making it easy to convert to an optional parameter.

WidgetsBurritos replied:
I can do so, but it is worth noting that the `create()` method below this always injects that service in anyway. So I'm not entirely sure we're buying much in terms of performance here?

bighappyface replied:
Not concerned about performance, just method signatures. Adding parameters into the middle tends to reflect an API break, and form classes extending this (although there really aren't many) would be impacted. Fair?

WidgetsBurritos replied:
Fair enough.  Pushing

WidgetsBurritos committed:
chore(parser): Swap order of parameters

WidgetsBurritos commented:
@bighappyface I addressed your feedback in a separate commit (this will get squashed when we merge the code): 

f6204d4ac71d6dc840b43cb4b1c59e9e9caeabdf

bighappyface reviewed (approved):
+1 for 988dd89
+1 for f6204d4
